presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/SimplifyExpressions.java

```diff
@@ -95,6 +100,11 @@ public PlanNode visitFilter(FilterNode node, RewriteContext<Void> context)
             if (simplified.equals(BooleanLiteral.TRUE_LITERAL)) {
                 return source;
             }
+            // TODO: this needs to check whether the boolean expression coerces to false in a more general way.
+            // E.g., simplify() not always produces a literal when the expression is constant
+            else if (BooleanLiteral.FALSE_LITERAL.equals(simplified) || simplified instanceof NullLiteral) {
```

Comment:
Can you rewrite the condition into ```simplified.equals(BooleanLiteral.FALSE_LITERAL) || simplified instanceof NullLiteral```. This way it is easier to read, you always compare simplified versus something else. 

Another option would be to use ```switch``` block here.

What do you think?